webapp/tests/Unit/Service/DomainServiceTest.php

```diff
@@ -51,4 +72,40 @@ public function testNewBoard__InternalServererror(): void
         $this->expectException(ServerExceptionInterface::class);
         $domainServiceClient->newBoard();
     }
+
+    /**
+     * @dataProvider provideRotateRemainingTileData
+     */
+    public function testRotateRemainingTile__Ok(string $boardJson, array $board, Rotation $rotation, int $expectedRotation): void
+    {
+        $mockResponse = new MockResponse($boardJson);
+        $mockClient = new MockHttpClient($mockResponse);
+
+        $domainServiceClient = new DomainService(
+            $mockClient,
+            "http://domain-api"
+        );
+
+        $updatedBoard = $domainServiceClient->rotateRemainingTile($board, $rotation);
+        $this->assertEquals($expectedRotation, $updatedBoard['remainingTile']['rotation']);
+    }
```

Comment:
Again, I'm not really sure these test provide much added value, since you need to mock what the Domain API does, and hence the remaining non-mocked amount of code is quite limited.

E2E tests would have much more added value here, but it would be easier if you had a way to force the initial board state.

Maybe this will be possible once you implement persistence into the app, but for now your E2E tests seem sufficient to me, and these Unit tests could be discarded.